perf-regression/gpu-margo-p2p-bw.cu

```diff
@@ -0,0 +1,796 @@
+            if (ret != 1) return (-1);
+            break;
+        case 'T':
+            ret = sscanf(optarg, "%d", &opts->threads);
+            if (ret != 1) return (-1);
+            break;
+        case 'D':
+            ret = sscanf(optarg, "%d", &opts->duration_seconds);
+            if (ret != 1) return (-1);
+            break;
+        case 't':
+            ret = sscanf(optarg, "%u,%u", &opts->mercury_timeout_client,
+                         &opts->mercury_timeout_server);
+            if (ret != 2) return (-1);
+            break;
+        case 'n':
+            opts->na_transport = strdup(optarg);
+            if (!opts->na_transport) {
+                perror("strdup");
+                return (-1);
+            }
+            break;
+        case 'm':
+            opts->mmap_filename = strdup(optarg);
+            if (!opts->mmap_filename) {
+                perror("strdup");
+                return -1;
+            }
+            break;
+        case 'a':
+            opts->align_buffer = 1;
+            break;
+
+	case 'j':
+            opts->gpu_to_mem = 1;
+            break;
+        case 'k':
+            opts->mem_to_gpu = 1;
+            break;
+
+        default:
+            return (-1);
+        }
+    }
+
+    if (opts->xfer_size < 1 || opts->concurrency < 1
+        || opts->duration_seconds < 1 || !opts->na_transport
+        || opts->warmup_seconds < 0) {
+        return (-1);
+    }
+
+    return (0);
+}
+
+static void usage(void)
+{
+    fprintf(
+        stderr,
+        "Usage: "
+        "margo-p2p-bw -x <xfer_size> -D <duration> -n <na>\n"
+        "\t-x <xfer_size> - size of each bulk tranfer in bytes\n"
+        "\t-D <duration> - duration of test in seconds\n"
+        "\t-n <na> - na transport\n"
+        "\t[-c <concurrency>] - number of concurrent operations to issue with "
+        "ULTs\n"
+        "\t[-T <os threads>] - number of dedicated operating system threads to "
+        "run ULTs on\n"
+        "\t[-d <filename>] - enable diagnostics output\n"
+        "\t[-t <client_progress_timeout,server_progress_timeout>] # use \"-t "
+        "0,0\" to busy spin\n"
+        "\t[-m <filename>] - use memory-mapped file as buffers instead of "
+        "malloc\n"
+        "\t[-X <xfer_memory>] - size of total memory buffer to allocate (and "
+        "iterate over during transfer) in each process\n"
+        "\t[-a] - explicitly align memory buffer to page size\n"
+        "\t[-w] - number of seconds to warm up before benchmark measurement\n"
+	"\t[-j] - transfer data from GPU memory of host A to main memory of host B\n"
+        "\t[-k] - transfer data from main memory of host A to GPU memory of host B\n\n"
+        "\tDefault without -j and -k options will transfer data\n"
+        "\tfrom GUP memory of host A to GPU memory of host B\n\n"
+        "\texample: mpiexec -n 2 ./margo-p2p-bw -x 4096 -D 30 -n verbs://\n"
+        "\t(must be run with exactly 2 processes\n");
```

Comment:
Since in this case we want to make sure we're transferring across nodes, we should probably only run when 2+ nodes are detected and we should split up MPI_COMM_WORLD to determine the ranks on the different nodes. I can make that improvement after this PR is merged.